components/director/internal/domain/subscription/service.go

```diff
@@ -401,13 +417,43 @@ func (s *service) DetermineSubscriptionFlow(ctx context.Context, providerID, reg
 	return "", errors.Errorf("could not determine flow")
 }
 
-func (s *service) createApplicationFromTemplate(ctx context.Context, appTemplate *model.ApplicationTemplate, subscribedSubaccountID, consumerTenantID, subscribedAppName, subdomain, region string) error {
+func (s *service) createApplicationFromTemplate(ctx context.Context, appTemplate *model.ApplicationTemplate, subscribedSubaccountID, consumerTenantID, subscribedAppName, subdomain, region string, subscriptionPayload string) error {
 	values := []*model.ApplicationTemplateValueInput{
-		{Placeholder: "name", Value: subscribedAppName},
-		{Placeholder: "display-name", Value: subscribedAppName},
 		{Placeholder: "subdomain", Value: subdomain},
 		{Placeholder: "region", Value: strings.TrimPrefix(region, RegionPrefix)},
 	}
+
+	oldPlaceholders := appTemplate.Placeholders
+
+	newPlaceholders := []model.ApplicationTemplatePlaceholder{}
+	for _, placeholder := range oldPlaceholders {
+		if placeholder.Name != "subdomain" && placeholder.Name != "region" {
+			newPlaceholders = append(newPlaceholders, placeholder)
+		}
+	}
+	appTemplate.Placeholders = newPlaceholders
+
+	appFromTemplateInput, err := s.appTemplateConv.ApplicationFromTemplateInputFromGraphQL(appTemplate, graphql.ApplicationFromTemplateInput{
+		TemplateName:        appTemplate.Name,
+		PlaceholdersPayload: &subscriptionPayload,
+	})
+
+	if err != nil {
+		return errors.Wrapf(err, "while parsing the callback payload with the Application template %q", appTemplate.Name)
+	}
+
+	appTemplate.Placeholders = oldPlaceholders
+
+	if len(subscriptionPayload) > 0 {
+		values = append(appFromTemplateInput.Values, values...)
+	} else {
+		additionalValues := []*model.ApplicationTemplateValueInput{
```

Comment:
I think this code in the `else` statement is never going to be reachable in an e2e flow, because in theory there should always be a subscriptionPayload + there's a GraphQL requirement that a `subscriptionPayload: String!` is provided

Reply:
Yes, I add this for compatibility when it was not mandatory, but will remove it now.